Import/Importer.cpp

```diff
@@ -270,35 +290,34 @@ static const char* get_row(const char* buf,
     } else if (*p == copy_params.delimiter || is_eol(*p, line_endings)) {
       if (!in_quote && !in_array) {
         if (!has_escape && !strip_quotes) {
-          std::string s = trim_space(field, p - field);
+          auto s = trim_space<String>(field, p - field);
           row.push_back(s);
         } else {
-          auto field_buf = std::make_unique<char[]>(p - field + 1);
           int j = 0, i = 0;
           for (; i < p - field; i++, j++) {
             if (has_escape && field[i] == copy_params.escape &&
                 field[i + 1] == copy_params.quote) {
-              field_buf[j] = copy_params.quote;
+              field[j] = copy_params.quote;
```

Comment:
You are modifying the given buffer (field( = buf )) instead of creating another buffer, which makes sense if we are sure that it will not used again. However if there is a problem in multithreaded calls, this buffer will be used from the beginning with a single thread. So I am guessing that the buffer must not be modified.

Reply:
AFAIK. Buffer is used just within a single thread, so it wouldn't cause a problem.

Reply:
> AFAIK. Buffer is used just within a single thread, so it wouldn't cause a problem.

Could you please check this function `Detector::split_raw_data()` ?